src/output-json-alert.c

```diff
@@ -173,6 +174,31 @@ static void AlertJsonSourceTarget(const Packet *p, const PacketAlert *pa,
     jb_close(js);
 }
 
+static void AlertJsonReference(
+        AlertJsonOutputCtx *json_output_ctx, const PacketAlert *pa, JsonBuilder *jb)
+{
+    if (!pa->s->references) {
+        return;
+    }
+
+    bool first = true;
+    const DetectReference *kv = pa->s->references;
+    while (kv) {
+        if (first) {
+            jb_open_array(jb, "references");
+            first = false;
+        }
+        const size_t size = strlen(kv->key) + strlen(kv->reference) + 1;
+        char kv_store[size];
```

Comment:
I do not like dynamic stack allocations...

Reply:
why not?

Reply:
Because small stacks were my hardest bugs like https://github.com/golang/go/issues/49075 or https://redmine.openinfosecfoundation.org/issues/6263

But I guess it is good here.

Do we have a maximum in the dev guide for such stack allocations ?

Reply:
I found that the values are capped (values from util-reference-config.h):
```
#define REFERENCE_SYSTEM_NAME_MAX   64
#define REFERENCE_CONTENT_NAME_MAX  1024
```

Thus, the maximum stack allocation would be 64 + 1024 + 1
```
char kv_store[REFERENCE_SYSTEM_NAME_MAX + REFERENCE_CONTENT_NAME_MAX + 1];
```

Reply:
I like it better, thanks :-)

Reply:
Perhaps
```
char kv_store[MIN(key_strlen,REFERENCE_SYSTEM_NAME_MAX) + MIN(ref_strlen,REFERENCE_CONTENT_NAME_MAX) + 1];
```

Reply:
or just a comment that these are bounded